programs/models.py

```diff
@@ -2,21 +2,51 @@
 from phonenumber_field.modelfields import PhoneNumberField
 from translations.models import Translation
 from programs.programs import calculators
+from programs.programs.fpl import FplCache
 from programs.util import Dependencies, DependencyError
 
+import requests
+from integrations.util.cache import Cache
+
+
+class FplCache(Cache):
+    expire_time = 60 * 60 * 24  # 24 hours
+    default = {}
+    api_url = "https://aspe.hhs.gov/topics/poverty-economic-mobility/poverty-guidelines/api/"
+    max_household_size = 8
+
+    # need to obtain the FPL for each of these years for hhSizes 1-8 and return that as a dictionary to match the current fpl dictionary
+    def update(self):
+        # need to obtain all of the years that we have data for on the backend
+        # we do this by querying the backend this needs to happen in update because anything below these functions won't run
+        fpls = FederalPoveryLimit.objects.filter(fpl__isnull=False).distinct()
+        fpl_dict = {}
+        for fpl in fpls:
+            household_sz_fpl = {}
+            # now we want to request the fpl from the api
+            for i in range(1, self.max_household_size + 1):
+                data = self._fetch_income_limit(fpl.period, str(i))
+                household_sz_fpl[i] = data
+                if i == self.max_household_size:
+                    income_limit_extra_member = self._fetch_income_limit(fpl.period, str(self.max_household_size + 1))
+                    household_sz_fpl["additional"] = income_limit_extra_member - data
+            fpl_dict[fpl.period] = household_sz_fpl
+        return fpl_dict
+
+    def _fetch_income_limit(self, year: str, household_size: str):
+        response = requests.get(self._fpl_url(year, household_size))
+        response.raise_for_status()  # Raise an exception for any unsuccessful request
```

Comment:
The exception comment is redundent.
```suggestion
    def _fetch_income_limit(self, year: str, household_size: str):
        '''
        Request the FPL from the API for a year and household size
        '''
        response = requests.get(self._fpl_url(year, household_size))
        response.raise_for_status()
```